Source/Src_nd/ppm.F90

```diff
@@ -1355,2216 +1354,2219 @@ subroutine ppm_2d(lo,hi,s,s_lo,s_hi,nc_s,&
+
+                !
+                ! Interpolate s to x-edges.
+                !
+                sp = HALF*(s(i+1,j,k,n)+s(i,j,k,n)) - SIXTH*(dsvl_r-dsvl_l)
+                !
+                ! Make sure sedge lies in between adjacent cell-centered values.
+                !
+                sp = max(sp,min(s(i+1,j,k,n),s(i,j,k,n)))
+                sp = min(sp,max(s(i+1,j,k,n),s(i,j,k,n)))
+
+                !
+                ! Modify using quadratic limiters.
+                !
+                if ((sp-s(i,j,k,n))*(s(i,j,k,n)-sm) .le. ZERO) then
+                   sp = s(i,j,k,n)
+                   sm = s(i,j,k,n)
+                else if (abs(sp-s(i,j,k,n)) .ge. TWO*abs(sm-s(i,j,k,n))) then
+                   sp = THREE*s(i,j,k,n) - TWO*sm
+                else if (abs(sm-s(i,j,k,n)) .ge. TWO*abs(sp-s(i,j,k,n))) then
+                   sm = THREE*s(i,j,k,n) - TWO*sp
                 end if
 
-                sigma = abs(u(i,j,k))*dt/dx(1)
-                s6 = SIX*s(i,j,k) - THREE*(sm+sp)
-                if (u(i,j,k) .lt. -rel_eps) then
-                   Im(i,j,k,1) = sm + &
-                        (sigma/TWO)*(sp-sm+(ONE-TWO3RD*sigma)*s6)
-                else
-                   Im(i,j,k,1) = s(i,j,k)
+                ! Different stencil needed for x-component of EXT_DIR and HOEXTRAP adv_bc's.
+                !
+                if (i .eq. lo(1)+1 .and. lo(1)+1 .eq. domlo(1)) then
+                   if (adv_bc(1,1,bccomp) .eq. EXT_DIR  .or. adv_bc(1,1,bccomp) .eq. HOEXTRAP) then
+
+                      !
+                      ! The value in the first cc ghost cell represents the edge value.
+                      !
+                      sm = s(i-1,j,k,n)
+                      !
+                      ! Use a modified stencil to get sedge on the first interior edge.
+                      !
+                      sedge = -FIFTH     *s(i-1,j,k,n) &
+                           + (THREE/FOUR)*s(i,j,k,n) &
+                           + HALF        *s(i+1,j,k,n) &
+                           - (ONE/20.0d0)*s(i+2,j,k,n)
+                      !
+                      ! Make sure sedge lies in between adjacent cell-centered values.
+                      !
+                      sedge = max(sedge,min(s(i+1,j,k,n),s(i,j,k,n)))
+                      sedge = min(sedge,max(s(i+1,j,k,n),s(i,j,k,n)))
+                      !
+                      ! Copy sedge into sp and sm.
+                      !
+                      sp = sedge
+
+                   end if
                 end if
 
-          else
+                if (i .eq. lo(1)+2 .and. lo(1)+1 .eq. domlo(1)) then
+                   if (adv_bc(1,1,bccomp) .eq. EXT_DIR  .or. adv_bc(1,1,bccomp) .eq. HOEXTRAP) then
+
+                      !
+                      ! Use a modified stencil to get sedge on the first interior edge.
+                      !
+                      sedge = -FIFTH     *s(i-2,j,k,n) &
```

Comment:
I don't think this stencil is right.  The idea is that we should never be using anything < domlo(1).  I think the first point should be `domlo(1)`.

Reply:
I think it would be clearer here and in the other parts to explicitly use domlo and domhi.

Reply:
all of these interior stencils below have the same issue

Reply:
Shouldn't it be that we shouldn't use anything < `lo(1)`? In this case, for grids on the boundary `lo(1)-1 = domlo(1)`